lib/devise/models/trackable.rb

```diff
@@ -46,6 +46,15 @@ def extract_ip_from(request)
         request.remote_ip
       end
 
+      private
+
+      def skip_trackable_and_not_active_for_authentication?(request)
+        request.env['devise.skip_trackable'] || not_active_for_authentication?
+      end
+
+      def not_active_for_authentication?
+        self.respond_to?(:active_for_authentication?) && !self.active_for_authentication?
```

Comment:
Can we ✂️ `self` here?
```suggestion
        respond_to?(:active_for_authentication?) && !active_for_authentication?
```

Reply:
Also, we don't need to ask if the resource `respond_to?(:active_for_authentication?)` since we already ask it in the caller.